feature/login/src/main/java/com/epmedu/animeal/login/code/domain/EnterCodeViewModel.kt

```diff
@@ -61,20 +61,29 @@ internal class EnterCodeViewModel @Inject constructor(
         }
     }
 
-    private fun validateCodeIfFull() {
-        if (state.code.all { it != null }) {
-            val isCodeCorrect = isCodeCorrect()
-            updateState { copy(isError = !isCodeCorrect) }
+    private fun validateCodeIfFilled() {
+        if (state.isCodeFilled()) {
+            validateCode()
+        }
+    }
 
-            viewModelScope.launch {
-                sendEvent(Event.NavigateToFinishProfile)
-            }
+    private fun validateCode() {
```

Comment:
The function does two things at once actually. From the name it is obvious it should validate a code, but not send a navigation event. Also I would rather put validation logic into init block, something like this:
```
 init {
        stateFlow
            .filterNotNull()
            .onEach {
                updateState {
                    copy(
                         isError = !isCodeEquals(CORRECT_CODE).also { isCorrect ->
                                 if (isCorrect) navigateToFinishProfile()
                         }
                    )
                }
            }
            .launchIn(viewModelScope)
    }
```

Reply:
Done